Added a new recipe for debezium source connector to Postgres

fmanimali opened the pull request:
# What changed, and why it matters
added a new file into terraform/cookbook

fmanimali commented:
thank you @laysauchoa for helping me with this pull request. Much appreciated!

fmanimali commented:
@dewan-ahmed the Linting/Vale  checks failures are no valid because the "kafka" and "postgresql" names are part of class names and cannot be changed.
Please merge. thanks.

dewan-ahmed commented:
Hi @fmanimali thanks for this PR. Since we already have [an existing recipe](https://developer.aiven.io/docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.html) to cover Kafka source connector, I'd like to understand the goal of having a similar recipe. The cookbook doesn't aim to cover all the combinations of one technology (for example, all Kafka source connectors), rather cover one or two unique recipes on multiple technologies. For example, [this PR](https://github.com/aiven/devportal/pull/1056) covers a second sink connector because it integrates with a third-party cloud provider. 

If you still feel that we need a second Kafka source connector, please let us know and we can happily assist you. Here are some pre-review comments on this PR:

1. The wording, story, diagram, and configuration should differ from each recipe. This recipe is too similar to [this recipe](https://developer.aiven.io/docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.html). 
2. We cannot merge PRs with failed checks. From a quick overview on the Linting/Vale, you can tackle a lot of them by doing what Vale suggests. For example, using "JSON" instead of "json", using literal for variable names in text (``defaultdb`` instead of "defaultdb), etc. One other thing is to use ".. code:: terraform" and format the code block properly so that the variables names are not treated as text. 

Thanks again for this contribution and looking forward to your response.

fmanimali commented:
Hello Dewan, you're right actually. I am doing this particular recipe
because a customer asked for it. I know that it's quite similar to others
but they asked for steps and instructions to setup a debezium/postgres with
kafka and it had to be multi cloud, infact kafka and postgres are in
different clouds.
Let me know if that's ok, otherwise we'll figure something different.

Thanks, F

On Mon, 4 Jul 2022, 18:54 Dewan Ishtiaque Ahmed, ***@***.***>
wrote:

> Hi @fmanimali <https://github.com/fmanimali> thanks for this PR. Since we
> already have an existing recipe
> <https://developer.aiven.io/docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.html>
> to cover Kafka source connector, I'd like to understand the goal of having
> a similar recipe. The cookbook doesn't aim to cover all the combinations of
> one technology (for example, all Kafka source connectors), rather cover one
> or two unique recipes on multiple technologies. For example, this PR
> <https://github.com/aiven/devportal/pull/1056> covers a second sink
> connector because it integrates with a third-party cloud provider.
>
> If you still feel that we need a second Kafka source connector, please let
> us know and we can happily assist you. Here are some pre-review comments on
> this PR:
>
>    1. The wording, story, diagram, and configuration should differ from
>    each recipe. This recipe is too similar to this recipe
>    <https://developer.aiven.io/docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.html>
>    .
>    2. We cannot merge PRs with failed checks. From a quick overview on
>    the Linting/Vale, you can tackle a lot of them by doing what Vale suggests.
>    For example, using "JSON" instead of "json", using literal for variable
>    names in text (defaultdb instead of "defaultdb), etc. One other thing
>    is to use ".. code:: terraform" and format the code block properly so that
>    the variables names are not treated as text.
>
> Thanks again for this contribution and looking forward to your response.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aiven/devportal/pull/1110#issuecomment-1174004130>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHHKWONRUFNL5UI52ASN23LVSMJNLANCNFSM52S4CFRA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mhoffm-aiven reviewed (dismissed):
* terraform manifest needs leading whitespace
* please make sure to select terraform lexer to make highlighting work properly

encima commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+
+The `Aiven Terraform Provider <https://registry.terraform.io/providers/aiven/aiven/latest/docs>`_ is a great choice for provisioning an Aiven for Apache Kafka® cluster with Kafka Connect enabled and the `Debezium source connector for PostgreSQL® <https://developer.aiven.io/docs/products/kafka/kafka-connect/howto/debezium-source-connector-pg.html>`_ configured.
+
+Let's check out the following diagram to understand the setup.
```
I would expand on this further. This is not only a recipe for Debezium but, on a higher level, how to integrate Kafka and Kafka Connect with services that may not be deployed in the same cloud.

encima commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+
+.. Warning::
+
+    Creating a `dedicated Aiven for Apache Kafka Connect service <https://developer.aiven.io/docs/products/kafka/kafka-connect/getting-started.html#apache-kafka-connect-dedicated-cluster>`_ is Aiven's suggested option. Having Kafka Connect running on the same nodes as Apache Kafka increases the load on the nodes possibly making the cluster more unstable. 
```
Be careful with wording here. Can having Kafka Connect on the same nodes always make the cluster more unstable? Is this Aiven's recommended option for all scenarios? 

> Aiven provides the option to run Kafka Connect on the same nodes as your Kafka cluster, sharing the resources. This is a low-cost way to get started with Kafka Connect but it is not recommended for Production environments, or scenarios where the deployed Connect cluster needs to be in a different cloud provider/region to the Kafka cluster. For these situations, Aiven recommends the use of a dedicated Aiven for Apache Kafka Connect service

encima reviewed (changes requested):
It might make sense for this to be a blog post more than a recipe if there is existing content but I still think Devportal fits as a point of reference. @dewan-ahmed this choice is up to you as you have better visibility of the dev documentation as a whole

mhoffm-aiven commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+
+The ``service.tf`` file for the provisioning af these 3 services and integrations is this:
+
+.. code::
```
please mark as terraform here for highlighting

laysauchoa replied:
does this work? I think is not supported

mhoffm-aiven replied:
i kinda randomly found that pygments supports terraform: https://pygments.org/docs/lexers/, we already use it i think for other recipes too

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+      maintenance_window_time = "10:00:00"
+      kafka_user_config {
+        kafka_rest      = true
+        kafka_connect   = false
```
I'm checking with Aiven Terraform Provider team on default value for this flag. In case it's ``false``, we don't need to explicitly set it.

fmanimali replied:
sometimes it adds to readability when properties are added even if they have default values. 
Just because our recies are also a way to show how things work not just execute them blindly. 
My personal experience is that having more properties in these examples help with understanding how things work.

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+      cloud_name              = "azure-norway-west"
+      plan                    = "startup-2"
+    #  service_name            = "${var.prefix}kf"
+      service_name            = "kf"
```
Same comment as above regarding general service names.

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+          min_insync_replicas          = 2
+        }
+
+        kafka_authentication_methods {
```
Are we using this authentication method for this recipe? If not, this can be removed.

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+      cloud_name              = "google-europe-north1"
+      project_vpc_id          = "francesco-demo/01a413b4-36df-4b1b-a697-fd7f87833494"
+      plan                    = "startup-4"
+    #  service_name            = "${var.prefix}kc"
```
Same comment as above regarding removing commented out lines.

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+      project_vpc_id          = "francesco-demo/01a413b4-36df-4b1b-a697-fd7f87833494"
+      plan                    = "startup-4"
+    #  service_name            = "${var.prefix}kc"
+      service_name            = "kc"
```
Same comment as above regarding general service names.

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+      source_service_name      = aiven_kafka.kf.service_name
+      destination_service_name = aiven_kafka_connect.kc.service_name
+
+      kafka_connect_user_config {
```
Are all the configurations needed for this setup? If yes, please add explanation on each setting/parameter at the end. If not, let's remove them.

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+        }
+      }
+
+      depends_on = [aiven_kafka_connect.kc,aiven_pg.demo]
```
Let's explain to the user why we're using https://www.terraform.io/language/meta-arguments/depends_on here. All explanations will go at the end (after the code block).

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+      service_name   = aiven_kafka_connect.kc.service_name
+      connector_name = "kafka-pg-source"
+
+      config = {
```
We have an [existing issue](https://github.com/aiven/devportal/issues/1118) to add explanation on configuration parameters. As good practice, let's add some explanation on the new recipes. This doesn't need to be exhaustive. But on a surface level, the audience should know why they are setting these parameters. The obvious ones like "database.hostname", "plugin.name", etc. don't need an explanation. But "snapshot.mode", "database.ssl.mode", etc. are good to explain a bit.

dewan-ahmed replied:
https://developer.aiven.io/docs/tools/terraform/reference/cookbook/kafka-custom-conf-recipe.html is a good one for inspiration.

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+  - The ``kafka_connect = false`` property is needed because we want to create a separate Aiven Apache Kafka Connect service.
+
+
+- One Aiven Apache Kafka Connect service is configured with public access
```
I suggest reviewing https://developer.aiven.io/docs/tools/terraform/reference/cookbook/kafka-custom-conf-recipe.html on how configuration options are explained.

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+
+- One Aiven Apache Kafka Connect service is configured with public access
+- Then a service integration is created within Kafka Connect service. This integration will use 2 internal topics for storing status and offset.
+- The last Aiven service that will be provisioned is the actual Debezium source connector for PostgreSQL, which is specified by the "connector.class" and is configured with the connection strings to access the PostgreSQL database and listen for all data changes on one or more tables. In our case, it will be "tab1" in "defaultdb", "public" schema. The plugin used is "wal2json" that converts WAL ev ...
```
A general note for all Vale warnings. Let's try to fix them. For example, "JSON" instead of "json", ``defaultdb`` instead of "defaultdb", etc. Please refer to https://github.com/aiven/devportal/blob/main/CONTRIBUTING.rst#formatting-guidelines for details. I'm happy to assist with any warning that you might be stuck with.

dewan-ahmed commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+
+Let's check out the following diagram to understand the setup.
+
+.. mermaid::
```
If we're creating resources across clouds, let's show that in the diagram. For example, the text could be "PostgreSQL service on GCP", "Apache Kafka on Azure".

dewan-ahmed reviewed (changes requested):
Hi @fmanimali I've added a few review comments and will be happy to review again after the next set of commits. Thank you.

fmanimali commented on _toc.yml:
```diff
@@ -144,6 +144,8 @@ entries:
             - file: docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe
               title: Apache Kafka MongoDB Source Connector
+            - file: docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source
+              title: Apache Kafka with Debezium Source Connector for PostgreSQL
```
cool, that sounds good, let me include all the changes and will commit a new version

fmanimali commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,197 @@
+
+    Be sure to check out the :doc:`getting started guide <../../get-started>` to learn about the common files required to execute the following recipe.
+    For example, you'll need to declare the variables for ``project``, ``api_token``.
+    These global variables that allow terraform to access the Aiven console and instruct deployments, are normally configured in a separate file called ``secret.tfvars`` and called 
```
repetita iuvant : I thought it wouldn't hurt to repeat some instructions :-)

encima reviewed (approved):
LGTM, will leave it to @dewan-ahmed to approve and merge based on his requested changes

mhoffm-aiven commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,195 @@
+        "name"            = "kafka-pg-source"
+        "connector.class" = "io.debezium.connector.postgresql.PostgresConnector",
+        "snapshot.mode"   = "initial"
+        "database.hostname" : aiven_pg.demo.service_host
```
formatting here looks kinda weird; can you do "=" and to "terrafmt" again?

fmanimali replied:
Michael, sorry I don't understand what change you are suggesting here

mhoffm-aiven replied:
"database.hostname" = aiven_pg.demo.service_host

mhoffm-aiven commented on docs/tools/terraform/reference/cookbook/kafka-debezium-postgres-source.rst:
```diff
@@ -0,0 +1,192 @@
+- The Aiven for Apache Kafka Connect service is configured with public access to allow the service to be accessed through a VPC since we are setting up services in different clouds
+- The resource "aiven_service_integration.i1" configures the integration between the AIven for Apache Kafka service and the Aiven for Apache Kafka Connect service. This integration uses 2 internal topics for storing status and offset.
+- The last Aiven service that will be provisioned is the Debezium source connector for PostgreSQL, which is specified by the "connector.class" and is configured with the connection strings to access the PostgreSQL database and listen for all data changes on one or more tables. In our case, the table that is monitored for any new data is "tab1" in "defaultdb", in "public" schema. The plugin used i ...
+- The "depends_on" property establishes a dependency between the services creation in order to avoid failures.
```
Depends on should not be used most of the time, its better to let TF do its thing, the only place where it is kinda needed ( and thats more a problem with our provider ) is in the connector to tell it to wait for the integration ( its an implicit dependency since creating the integration starts kafka connect up and we have to wait for it)

fmanimali commented:
removed VPC, and corrected the "depends_on = [aiven_service_integration.i1]". I had increased the number of resources in depends_on because after some tests I had seen the 503 error again. Now it's gone.

mhoffm-aiven commented:
> removed VPC, and corrected the "depends_on = [aiven_service_integration.i1]". I had increased the number of resources in depends_on because after some tests I had seen the 503 error again. Now it's gone.

yeah! sadly thats just a bit of residual flakiness. what i wanted to say is: adding the depends_on does not do anything since terraform already knows that dependency since the resource was referenced.

encima commented:
> removed VPC, and corrected the "depends_on = [aiven_service_integration.i1]". I had increased the number of resources in depends_on because after some tests I had seen the 503 error again. Now it's gone.

VPC is needed. The aim of this post is to show connectivity between services in multi cloud environments where those services are not all available over the public internet. I.e. MongoDB/MySQL etc in a VPC 

![image](https://user-images.githubusercontent.com/517923/178996293-a538c80c-391e-4d9e-985a-d6641ce23044.png)

encima commented:
without the vpc would work but the aim of the test is to show it with the vpc. it is a requirement in the terraform script

( https://www.facebook.com/aivencloud ) ( https://twitter.com/aiven_io ) ( https://www.linkedin.com/in/christopher-gwilliams-51322924/ )

Dr. Chris Gwilliams

Staff Solution Architect

Solution Architecture | Aiven

+358444915385

***@***.***

aiven.io ( http://aiven.io )

Helsinki, Finland

On Thu, Jul 14 2022 at 7:17 PM, Francesco Animali < ***@***.*** > wrote:

> 
> 
> 
> 
> 
> tbf I have just tested without the vpc_id in the terraform service:
> 
> ```
> resource "aiven_kafka_connect" "kc" {
> project = var.project
> cloud_name = "google-europe-north1"
> # project_vpc_id =
> "francesco-demo/01a413b4-36df-4b1b-a697-fd7f87833494"
> plan = "startup-4"
> ```
> 
> and the integration of kafka connect and postgres works just fine. I still
> 
> have the vpc defined in the project , but I am not referencing the vpc
> inside the terraform script.
> 
> On Thu, Jul 14, 2022 at 3:42 PM Chris Gwilliams ***@***.***>
> wrote:
> 
> > removed VPC, and corrected the "depends_on =
> > [aiven_service_integration.i1]". I had increased the number of resources
> in
> > depends_on because after some tests I had seen the 503 error again. Now
> > it's gone.
> >
> > VPC is needed. The aim of this post is to show connectivity between
> > services in multi cloud environments where those services are not all
> > available over the public internet. I.e. MongoDB/MySQL etc in a VPC
> >
> > [image: image]
> >
> <https://user-images.githubusercontent.com/517923/178996293-a538c80c-391e-4d9e-985a-d6641ce23044.png>
> 
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/aiven/devportal/pull/1110#issuecomment-1184464900>,
> > or unsubscribe
> >
> <https://github.com/notifications/unsubscribe-auth/AHHKWOP3V6XZJ6NALS5UAIDVUAKLZANCNFSM52S4CFRA>
> 
> > .
> > You are receiving this because you were mentioned.Message ID:
> > ***@***.***>
> >
> 
> 
> --
> [image: Aiven] <https://www.aiven.io/>
> *Francesco Animali*
> Solutions architect, *Aiven*
> ***@***.*** ***@***.***>
> aiven.io <https://www.aiven.io/> |
> <https://www.facebook.com/aivencloud>
> <https://www.linkedin.com/company/aiven/> <https://twitter.com/aiven_io>
> 
> 
> —
> Reply to this email directly, view it on GitHub (
> https://github.com/aiven/devportal/pull/1110#issuecomment-1184699579 ) , or
> unsubscribe (
> https://github.com/notifications/unsubscribe-auth/AAD6OIYRPJEX4TNLPM5BMK3VUBDSXANCNFSM52S4CFRA
> ).
> You are receiving this because your review was requested. Message ID: <aiven/devportal/pull/1110/c1184699579
> @ github. com>
> 
> 
>

dewan-ahmed commented:
Hi @fmanimali I added the following changes via https://github.com/aiven/devportal/pull/1110/commits/bbd805191fdaedde99373400eb7722f6f81ba8ce:
1. Improved the Kafka Connect warning text based on a discussion with @encima 
2. Fixed Vale warnings and minor grammar issues
3. Removed the additional instruction for secrets.tfvars since https://github.com/aiven/devportal/issues/959 will take care of the common files. 

The PR is really close to completion. Please resolve the following issues:
1. Check about whether to keep/discard VPC in this recipe. I don't have a particular preference on this - whatever you and Chris decide.
2. Please rename the Terraform resources to something more descriptive and uniform with other recipes. For example, ``demo-pg`` instead of ``demo`` and ``demo-kafka`` instead of ``kf``, etc. Please check out the existing recipes for naming practices.
3. Please remove the ``depends_on`` references since those are not used.
4. No need to mention "Version 3.2.1 of the Aiven Terraform provider will be used", "The PostgreSQL database will be created in "google-europe-north1" etc. since this is obvious to the reader. Please explain the configuration parameters and their functions. For example, ``group_id``, ``offset_storage_topic``, ``status_storage_topic`` under ``kafka_connect_user_config`` which is part of the ``aiven_service_integration``

Right now, the Sphinx/prose check is failing due to a broken link. Don't worry about that as the 
`(docs/integrations/rsyslog: line  145) broken    ./datadog.html` file is not related to this PR.

Please let me know once the changes are in and I'll review. Thanks.

fmanimali commented:
@mhoffm-aiven not sure why the "prose" check is failing. Is it something I should fix in the document?

mhoffm-aiven commented:
> @mhoffm-aiven not sure why the "prose" check is failing. Is it something I should fix in the document?

```
(docs/integrations/rsyslog: line  145) broken    ./datadog.html - 
```
not your fault, something on main i think

fmanimali commented:
@mhoffm-aiven all requested changes are in.

dewan-ahmed commented:
> @dewan-ahmed kindly review, good to be published for me. thanks :-)

thanks @fmanimali 
I'm travelling to Europe for a conference. Review might be delayed. If I cannot review within my travel time, I'll review this Friday when I'm back.

dewan-ahmed commented:
Hi @fmanimali I reviewed the PR but I still see the existing issues that I mentioned in my previous comment. Your last commit  https://github.com/aiven/devportal/pull/1110/commits/cdcd0584ca63515c0574b32049e45817b49a5d8a updated ":" to "=" for some parameters. 

I'm copying the requested changes from previous comment for reference:

1. Please use a descriptive Terraform resource name and Aiven service name. Update "demo" to "demo-pg", "kf" to "demo-kafka", "kc" to "demo-kafka-connect", etc. 
2. Are you using or dropping VPC in this recipe? If this recipe does not use VPC, please remove the parameter ``project_vpc_id`` and the explanation about VPC. If you're keeping VPC, please remove personal name like "francesco-demo" and replace with a variable. Example: **var.project_vpc_id** and we tell the user to declare this variable beforehand.
3. Please remove the depends_on references since those are not used.
4. No need to mention "Version 3.2.1 of the Aiven Terraform provider will be used", "The PostgreSQL database will be created in "google-europe-north1", "The Aiven Apache Kafka service will be created in ..." etc. since this is obvious to the reader. Please explain the configuration parameters and their functions. For example, group_id, offset_storage_topic, status_storage_topic under kafka_connect_user_config which is part of the aiven_service_integration.
5. Please re-run the Terraform script after these changes to ensure that the services are still created correctly.

Please request a review after you add these changes. Thanks.

mhoffm-aiven commented:
> Please remove the depends_on references since those are not used.

One is needed ( connector depends on integration ), otherwise i agree

fmanimali commented:
Hi Dewan, 

thank you for your thorough review. I have implemented all your changes and committed .
i left the vpc because it's needed. Also left depends_on the i1 integration because it's needed too.
Changed the name of "project_vpc_id" to something generic because it contained my name.

dewan-ahmed commented:
Thanks @fmanimali 
I did some small clean-ups via https://github.com/aiven/devportal/pull/1110/commits/a1fff40b84493e710b87b9fefe9203fcc48d95f3 (for example, there was a comma within the connector configs).

The only thing left to test in this PR is running the Terraform code and making sure that the services are created/configured correctly.
<img width="1319" alt="Screen Shot 2022-07-22 at 1 12 08 PM" src="https://user-images.githubusercontent.com/30604461/180590519-b39c0676-a0d4-425f-9727-be4bcd7039f0.png">
^^ This is where I'm having some issues.

I'm meeting Felix on Tuesday to test the VPC setup and then I can try this in Terraform. In case you can validate this setup before that time, please let me know. I also feel that we need to add some reference to VPC creation and VPC peering for the audience.

dewan-ahmed commented:
Hi @fmanimali I'm able to create the VPC and the VPC peering. Is the expectation that if I create a table called `tab1` under `defaultdb` and insert some data, I'll see a topic created with the name "someName-tab1" on the Kafka cluster on Azure? I don't see this behavior with this setup. Could you please confirm on the testing?

fmanimali commented:
> 



> created with the name "someName-tab1" on the Kafka cluster on Azure?

Dewan, yes, the expectation is that Debezium will instruct kafka to create a topic called "replicator.public.tab1". This is explained in the article, let me know if taht is not clear.

dewan-ahmed reviewed (approved):
Thanks @fmanimali for your patience and this great contribution to the Terraform cookbook 🎉

ftisiot commented:
If you rebase  the branch with main, the error should be gone

> On 18 Jul 2022, at 13:06, Michael Hoffmann ***@***.***> wrote:
> 
> 
> @mhoffm-aiven <https://github.com/mhoffm-aiven> not sure why the "prose" check is failing. Is it something I should fix in the document?
> 
> (docs/integrations/rsyslog: line  145) broken    ./datadog.html - 
> not your fault, something on main i think
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/aiven/devportal/pull/1110#issuecomment-1187087312>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACWAEJ6A2ARMBQTOTW56NRTVUU3CTANCNFSM52S4CFRA>.
> You are receiving this because you are subscribed to this thread.
>
